File: backend/src/main/java/com/tododuk/domain/todoLabel/repository/TodoLabelRepository.java
package com.tododuk.domain.todoLabel.repository;

import com.tododuk.domain.todoLabel.entity.TodoLabel;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface TodoLabelRepository extends JpaRepository<TodoLabel, Integer> {

    List<TodoLabel> findByTodoId(int todoId);

    Optional<TodoLabel> findByTodoIdAndLabelId(int todoId, int labelId);

    boolean existsByTodoIdAndLabelId(int todoId, int labelId);
}

File: backend/src/main/java/com/tododuk/domain/todoLabel/repository/TodoLabelRepository.kt
package com.tododuk.domain.todoLabel.repository

import com.tododuk.domain.todoLabel.entity.TodoLabel
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface TodoLabelRepository : JpaRepository<TodoLabel, Int> {
    fun findByTodoId(todoId: Int): MutableList<TodoLabel>

    fun findByTodoIdAndLabelId(todoId: Int, labelId: Int): Optional<TodoLabel>

    fun existsByTodoIdAndLabelId(todoId: Int, labelId: Int): Boolean
}